tests/test_filterDiaSourceCatalog.py

```diff
@@ -115,6 +137,37 @@ def test_run_with_filter_sky_only(self):
         self.assertEqual(len(result.rejectedDiaSources), self.nSkySources)
         self.assertEqual(len(self.diaSourceCat), self.nSources)
 
+    def test_run_with_filter_negative_only(self):
+        """Test that when only the negative filter is turned on then
+        sources which below the negtive snr cut are filtered out of the
+        catalog and the rest are returned.
+        """
+        self.config.doRemoveSkySources = False
+        self.config.doRemoveNegativeDirectImageSources = True
+        self.config.doWriteRejectedSkySources = True
+        self.config.doTrailedSourceFilter = False
+        filterDiaSourceCatalogTask = FilterDiaSourceCatalogTask(config=self.config)
+        result = filterDiaSourceCatalogTask.run(self.diaSourceCat, self.visitInfo)
+        nExpectedFilteredSources = self.nSources - self.nRemovedNegativeSources
+        self.assertEqual(len(result.filteredDiaSourceCat), nExpectedFilteredSources)
+        self.assertEqual(len(result.rejectedDiaSources), self.nRemovedNegativeSources)
+        self.assertEqual(len(self.diaSourceCat), self.nSources)
```

Comment:
Perhaps also add an assert that `self.diaSourceCat["scienceFlux"]/self.diaSourceCat["scienceFluxErr"]` contains `self.nRemovedNegativeSources` sources below the threshold, and `result.filteredDiaSourceCat` contains none below the threshold.